src/UpdatePaymentMethod/UpdatePaymentMethodBuilder.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+
+namespace Laravel\Cashier\UpdatePaymentMethod;
+
+use Illuminate\Database\Eloquent\Model;
+use Laravel\Cashier\FirstPayment\Actions\AddBalance;
+use Laravel\Cashier\FirstPayment\FirstPaymentBuilder;
+use Laravel\Cashier\Plan\Contracts\PlanRepository;
+use Laravel\Cashier\UpdatePaymentMethod\Contracts\UpdatePaymentMethodBuilder as Contract;
+
+class UpdatePaymentMethodBuilder implements Contract
+{
+    /**
+     * The billable model.
+     *
+     * @var \Illuminate\Database\Eloquent\Model
+     */
+    protected $owner;
+
+    /**
+     * UpdatePaymentMethodBuilder constructor.
+     *
+     * @param \Illuminate\Database\Eloquent\Model $owner
+     */
+    public function __construct(Model $owner)
+    {
+        $this->owner = $owner;
+    }
+
+    /**
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function submit()
+    {
+        $payment = (new FirstPaymentBuilder($this->owner))
+            ->setRedirectUrl(config('cashier.update_payment_method.redirect_url'))
+            ->setFirstPaymentMethod($this->allowedPaymentMethod())
+            ->inOrderTo($this->addToBalanceAction())
```

Comment:
depending on a config flag, either `addToBalance` or `addGenericItem`.

Reply:
See Spark for example: https://github.com/laravel/spark-aurelius-mollie/blob/986a84d3753944c8bd132a7e3b44168915cdecee/src/Interactions/Settings/PaymentMethod/UpdateMolliePaymentMethod.php#L45-L63

Reply:
> depending on a config flag, either `addToBalance` or `addGenericItem`.

if we do that, we need to do it for the first payment

Reply:
How about by default we add a generic order item, and add an `addToBalance()` method to the `UpdatePaymentMethodBuilder` - instead of using the config? Feels better to me.

```php
$billable
    ->updatePaymentMethod() // returns builder
    ->addToBalance()
    ->create();
```

Reply:
Ofcourse we can also do this on the FirstPaymentMethod, and wrap that in the UpdatePAymentMethodBuilder

Reply:
Updated.

Now the code will be like this:
```php
$billable
    ->updatePaymentMethod() 
    ->addGenericItem() // will add a generic Item
    ->create();
```
and by default will add credit, without use `addGenericItem()`

```php
$billable
    ->updatePaymentMethod() 
    ->create();
```